compiler/rustc_metadata/src/rmeta/encoder.rs

```diff
@@ -1547,6 +1547,10 @@ impl<'a, 'tcx> EncodeContext<'a, 'tcx> {
 
             record_defaulted_array!(self.tables.module_children_reexports[def_id] <-
                 module_children.iter().filter(|child| !child.reexport_chain.is_empty()));
+
+            let ambiguity_module_children = tcx.ambiguity_module_children_local(local_def_id);
+
+            record_defaulted_array!(self.tables.ambiguity_module_children[def_id] <- ambiguity_module_children.iter());
```

Comment:
```suggestion
            record_defaulted_array!(self.tables.ambiguity_module_children[def_id] <- ambiguity_module_children);
```
`lazy_array` takes an `IntoIterator`.

Nit: also there's a useless empty line above.